src/daemon/daemon.cpp

```diff
@@ -1480,11 +1480,14 @@ mp::Daemon::Daemon(std::unique_ptr<const DaemonConfig> the_config)
         allocated_mac_addrs = std::move(new_macs); // Add the new macs to the daemon's list only if we got this far
 
         // FIXME: somehow we're writing contradictory state to disk.
-        if (spec.deleted && spec.state != VirtualMachine::State::stopped)
-        {
-            mpl::log(mpl::Level::warning, category,
-                     fmt::format("{} is deleted but has incompatible state {}, resetting state to 0 (stopped)", name,
-                                 static_cast<int>(spec.state)));
+        if (spec.deleted && spec.state > VirtualMachine::State::stopped)
```

Comment:
This is now reliant on the relative positioning of states in the State enum. Wouldn't checking for the Off state be better?

The enum order might be why the incoherent log was being printed in the first place.

Reply:
Probably, yeah... OK, I'll fix it.

Reply:
Done!